iym/tasks.py

```diff
@@ -0,0 +1,349 @@
+            data_package_name,
+            datastore_token,
+        )
+
+        data_package_upload_authorisation = authorise_data_package_upload_result[
+            "filedata"
+        ]["data_package.json"]
+        upload(data_package_path, data_package_upload_authorisation)
+        update_import_report(
+            obj_to_update,
+            f'Datapackage url: {data_package_upload_authorisation["upload_url"]}',
+        )
+
+    return data_package_upload_authorisation
+
+
+def import_uploaded_package(data_package_url, datastore_token, obj_to_update):
+    import_query = {"datapackage": data_package_url, "jwt": datastore_token}
+    import_url = (
+        f"{settings.OPENSPENDING_HOST}/package/upload?{urlencode(import_query)}"
+    )
+    r = requests.post(import_url)
+    update_import_report(obj_to_update, f"Initial status: {r.text}")
+
+    r.raise_for_status()
+    status = r.json()["status"]
+
+    return status
+
+
+def update_import_report(obj_to_update, message):
+    now = datetime.datetime.now().strftime("%H:%M:%S")
+    obj_to_update.import_report += f"{now} - {message}" + os.linesep
+    obj_to_update.save()
+
+
+def check_and_update_status(status, data_package_url, obj_to_update):
+    last_logged_progress = -1
+    status_query = {
+        "datapackage": data_package_url,
+    }
+    status_url = (
+        f"{settings.OPENSPENDING_HOST}/package/status?{urlencode(status_query)}"
+    )
+    update_import_report(
+        obj_to_update, f"Monitoring status until completion ({status_url}):"
+    )
+    while status not in ["done", "fail"]:
+        time.sleep(5)
+        r = requests.get(status_url)
+        r.raise_for_status()
+        status_result = r.json()
+        new_progress = int(float(status_result["progress"]) * 100)
+        if new_progress != last_logged_progress:
+            update_status(
+                obj_to_update,
+                f"loading data ({new_progress}%)",
+            )
+            update_import_report(
+                obj_to_update,
+                f"loading data ({new_progress}%)",
+            )
+            last_logged_progress = new_progress
+        status = status_result["status"]
+
+        if status == "fail":
+            print(status_result["error"])
+
+    update_status(obj_to_update, status)
+
+
+def update_status(obj_to_update, status):
+    obj_to_update.status = status
+    obj_to_update.save()
+
+
+def process_uploaded_file(obj_id):
+    # read file
+    obj_to_update = models.IYMFileUpload.objects.get(id=obj_id)
+    if obj_to_update.process_completed:
+        return
+    try:
+        update_status(obj_to_update, "process started")
+        update_import_report(obj_to_update, "Cleaning CSV")
+
+        financial_year = obj_to_update.financial_year.slug
+        userid = settings.OPEN_SPENDING_USER_ID
+        data_package_name = f"national-in-year-spending-{financial_year}"
+        data_package_title = f"National in-year spending {financial_year}"
```

Comment:
for 
![Screenshot_2023-07-06_14-15-16](https://github.com/vulekamali/datamanager/assets/235801/71ef91b5-2243-4382-95c4-1b9b75e685bf)

you could define a new env var `OPENSPENDING_DATASET_CREATE_SUFFIX` defaulting to empty string in settings.py, default to `-dev` in docker-compose, and in sandbox we can configure it to be `-sandbox`

```suggestion
        data_package_name = f"national-in-year-spending-{financial_year}{settings.OPENSPENDING_DATASET_CREATE_SUFFIX}"
        data_package_title = f"National in-year spending {financial_year} {settings.OPENSPENDING_DATASET_CREATE_SUFFIX}"
```

Reply:
done